lib/BadgeOverlaySupport/BadgeOverlaySupport.js

```diff
@@ -0,0 +1,260 @@
+
+	/**
+	* Url for icon to be used in place of default icon.
+	*
+	* @name moon.BadgeOverlaySupport#badgeSrc
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+	badgeSrc: undefined,
+
+	/**
+	* Sepcifies the alignment position of badge.
+	* Possible values are 'top' or 'bottom'.
+	* Defaulted as 'bottom'.
+	*
+	* @name moon.BadgeOverlaySupport#position
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+	position: undefined,
+
+	/**
+	* Sepcifies the alignment position of badge icon.
+	* Possible values are 'left' or 'right'.
+	* Defaulted as 'left'.
+	*
+	* @name moon.BadgeOverlaySupport#badgeIconPosition
+	* @type {String}
+	* @default left
+	* @public
+	*/
+	badgeIconPosition: 'left',
+
+	/**
+	* Sepcifies whether badge should be transparent or not.
+	* Possible values are 'true' or 'false'.
+	* Defaulted as 'false'.
+	*
+	* @name moon.BadgeOverlaySupport#transparency
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+	transparency: false,
+
+
+	/**
+	* @private
+	*/
+	events: {
+		onBadgeTap: '',
+		onBadgeIconTap: ''
+	},
+
+	/**
+	* @method
+	* @private
+	*/
+	create: kind.inherit(function (sup) {
+		return function () {
+			this.kindComponents = enyo.Component.overrideComponents(this.kindComponents, this._badgeOverride);
+			sup.apply(this, arguments);
+			this.positionChanged();
+			this.badgeTypeChanged();
+			this.badgeIconPositionChanged();
+			if (this.badgeSrc) {
+				this.$.badgeScrimIcon.set('icon','');
+			}
+			this.transparencyChanged();
+		};
+	}),
+
+	/**
+	* @private
+	*/
+	bindings: [
+		{from: 'badgeSrc', to: '$.badgeScrimIcon.src'},
+		{from: 'badgeIcon', to: '$.badgeScrimIcon.icon', transform: function(val){ return val ? val : 'play';}}
+	],
+
+	/**
+	* @private
+	*/
+	_badgeOverride: {
+		image: {kind: Image, components:[{
+				name: 'badgeScrimIcon', kind: Icon, small: true, icon: 'play', spotlight: false
+			}]
+		}
+	},
+
+	/**
+	* @private
+	*/
+	positionChanged: function () {
+		this.addRemoveClass('top', this.position == 'top');
+	},
+
+	/**
+	* @private
+	*/
+	badgeIconPositionChanged: function() {
+		this.addRemoveClass('right', this.badgeIconPosition == 'right');
+	},
+
+	/**
+	* @private
+	*/
+	badgeTypeChanged: function () {
+		var image = this.$.image;
+		image.destroyClientControls();
+		if(this.badgeType && this.badgeType != 'image') {
+			image.createComponent(this._badgeOverride.image.components[0]);
+		}
+	},
+
+	/**
+	* @private
+	*/
+	transparencyChanged: function() {
+		this.addRemoveClass('moon-badge-overlay-transparent', this.transparency);
+	},
+
+	/**
+	* @method
+	* @private
+	*/
+	dispatchEvent: enyo.inherit(function (sup) {
+		return function (sEventName, oEvent, oSender) {
+			if (oEvent && !oEvent.delegate) {
+				if (sEventName == 'ontap') {
+					if (this.badgeTap(oSender, oEvent)) {
+						return true;
+					}
+				}
+			}
+			return sup.apply(this, arguments);
+		};
+	}),
+
+	/**
+	* @method
+	* @private
+	*/
+	badgeTap: function (inSender, inEvent) {
```

Comment:
Minor nitpick: Can we update the parameter names such that they are not prefixed with "in", i.e. `(sender, ev)`?